build.rs

```diff
@@ -1,20 +1,37 @@
 use std::env;
 
 fn main() {
-    if env::var("HOST") != env::var("TARGET") {   // assume cross-compiling for Kobo
+    let quadruple = env::var("TARGET").unwrap();
+    if quadruple == "arm-unknown-linux-gnueabihf" { //=> assume cross-compiling for Kobo
         println!("cargo:rustc-env=PKG_CONFIG_ALLOW_CROSS=1");
+
         println!("cargo:rustc-link-search=src/wrapper/Kobo");
         println!("cargo:rustc-link-search=libs");
-    } else {                                      // assume compiling for Linux host
-        println!("cargo:rustc-link-search=src/wrapper/Linux");
-        println!("cargo:rustc-link-lib=lcms2");
-        println!("cargo:rustc-link-lib=mupdf-third");
+
+        println!("cargo:rustc-link-lib=dylib=stdc++");
+
+    } else {
+        let tokens: Vec<&str> = quadruple.split('-').collect();
+        // [0] is <arch><sub> | [1] is <vendor> | [2] is <sys> | [3] is <abi>
+        let sys = tokens[2];
```

Comment:
You can directly use the `CARGO_CFG_TARGET_OS` environment variable.

Reply:
Nice!